src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -162,57 +115,35 @@ public static void MoveFile(string sourceFullPath, string destFullPath, bool ove
         {
             // The desired behavior for Move(source, dest) is to not overwrite the destination file
             // if it exists. Since rename(source, dest) will replace the file at 'dest' if it exists,
-            // link/unlink are used instead. Rename is more efficient than link/unlink on file systems
-            // where hard links are not supported (such as FAT). Therefore, given that source file exists,
-            // rename is used in 2 cases: when dest file does not exist or when source path and dest
-            // path refer to the same file (on the same device). This is important for case-insensitive
+            // link/unlink are used instead. rename is used when the source path and dest path refer
+            // to the same file (on the same device). This is important for case-insensitive
             // file systems (e.g. renaming a file in a way that just changes casing), so that we support
-            // changing the casing in the naming of the file. If this fails in any way (e.g. source file
-            // doesn't exist, dest file doesn't exist, rename fails, etc.), we just fall back to trying the
-            // link/unlink approach and generating any exceptional messages from there as necessary.
-            Interop.Sys.FileStatus sourceStat, destStat;
-            if (Interop.Sys.LStat(sourceFullPath, out sourceStat) == 0 && // source file exists
-                (Interop.Sys.LStat(destFullPath, out destStat) != 0 || // dest file does not exist
-                (sourceStat.Dev == destStat.Dev && // source and dest are on the same device
-                sourceStat.Ino == destStat.Ino)) && // source and dest are the same file on that device
-                Interop.Sys.Rename(sourceFullPath, destFullPath) == 0) // try the rename
-            {
-                // Renamed successfully.
-                return;
-            }
+            // changing the casing in the naming of the file.
 
-            // If overwrite is allowed then just call rename
-            if (overwrite)
+            // Note that the test is actually in native code, and the only thing managed code does
+            // is check fo EXDEV and call CopyFile.
+            if (Interop.Sys.Rename(sourceFullPath, destFullPath, overwrite ? 1 : 0) < 0)
```

Comment:
The 0 value for the flags is not defined in the comments above the P/Invoke method Interop.Sys.Rename. Consider mentioning what it means in the comments.

Reply:
Can do. It's a flags enum. I'll think of good wording soon.

Reply:
I think I improved the comment enough. If not, I'll have to start commenting each call site.